pydantic/fields.py

```diff
@@ -402,7 +402,7 @@ def merge_field_infos(*field_infos: FieldInfo, **overrides: Any) -> FieldInfo:
             field_info._attributes_set.update(overrides)
             for k, v in overrides.items():
                 setattr(field_info, k, v)
-            return field_info
+            return field_info  # type: ignore
```

Comment:
Why the ignore here?

Reply:
I think this set of changes should not be done — there's no need for this to be a classmethod (it isn't using `cls`), and you aren't guaranteed to get an instance of `Self`. You'll generally get whatever the type of the first element of the `field_infos` argument is, which is annotated as `FieldInfo`. So the best you can say is it will return a `FieldInfo`.

Reply:
Oh I see you changed it to use cls below. But do we need that? Have we subclassed `FieldInfo` anywhere? I think better to use a staticmethod given that we can't guarantee you'd receive an instance of the subclass.

If you want to change that, you should make `field_infos: typing_extensions.Self` as well.

Reply:
I think it will be easier to revert to `FieldInfo` instead of `Self` for this class. As I said these methods either return `Self` (by instanciating `cls`) or explicit `FieldInfo` instances, so not worth the overhead considering it isn't really public API